test/linearAlgebra/linearSystem.cpp

```diff
@@ -37,14 +37,74 @@ TEST_CASE("LinearSystem")
         );
 
         REQUIRE(linearSystem.matrix().values().size() == 9);
-        REQUIRE(linearSystem.matrix().nNonZeros() == 9);
+        REQUIRE(linearSystem.matrix().colIdxs().size() == 9);
         REQUIRE(linearSystem.matrix().rowPtrs().size() == 4);
         REQUIRE(linearSystem.matrix().nRows() == 3);
-
         REQUIRE(linearSystem.rhs().size() == 3);
     }
 
 
+    SECTION("view read/write " + execName)
+    {
+        NeoFOAM::Field<NeoFOAM::scalar> values(exec, {1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0});
+        NeoFOAM::Field<NeoFOAM::localIdx> colIdx(exec, {0, 1, 2, 0, 1, 2, 0, 1, 2});
+        NeoFOAM::Field<NeoFOAM::localIdx> rowPtrs(exec, {0, 3, 6, 9});
+        NeoFOAM::la::CSRMatrix<NeoFOAM::scalar, NeoFOAM::localIdx> csrMatrix(
+            values, colIdx, rowPtrs
+        );
+        NeoFOAM::Field<NeoFOAM::scalar> rhs(exec, {10.0, 20.0, 30.0});
+        NeoFOAM::la::LinearSystem<NeoFOAM::scalar, NeoFOAM::localIdx> ls(csrMatrix, rhs, "custom");
+
+        auto lsView = ls.view();
+        auto hostLS = ls.copyToHost();
+        auto hostLSView = hostLS.view();
+
+
+        // some simple sanity checks
+        REQUIRE(hostLSView.A.value.size() == 9);
+        REQUIRE(hostLSView.A.columnIndex.size() == 9);
+        REQUIRE(hostLSView.A.rowOffset.size() == 4);
+        REQUIRE(hostLSView.b.size() == 3);
+
+        // check system values
+        for (size_t i = 0; i < hostLSView.A.value.size(); ++i)
+        {
+            REQUIRE(hostLSView.A.value[i] == static_cast<NeoFOAM::scalar>(i + 1));
+            REQUIRE(hostLSView.A.columnIndex[i] == (i % 3));
+        }
+        for (size_t i = 0; i < hostLSView.A.rowOffset.size(); ++i)
+        {
+            REQUIRE(hostLSView.A.rowOffset[i] == static_cast<NeoFOAM::localIdx>(i * 3));
+        }
+        for (size_t i = 0; i < hostLSView.b.size(); ++i)
+        {
+            REQUIRE(hostLSView.b[i] == static_cast<NeoFOAM::scalar>((i + 1) * 10));
+        }
+
+        // Modify values.
+        parallelFor(
+            exec,
+            {0, lsView.A.value.size()},
+            KOKKOS_LAMBDA(const size_t i) {
+                lsView.A.value[i] = -lsView.A.value[i];
+                lsView.b[i] = -lsView.b[i];
+            }
+        );
```

Comment:
this does not work since` A.value` and `b` have size `nnz` and `ncells` respectively.
```suggestion
        // Modify values.
        parallelFor(
            exec,
            {0, lsView.A.value.size()},
            KOKKOS_LAMBDA(const size_t i) {
                lsView.A.value[i] = -lsView.A.value[i];
            }
        );
        
         // Modify values.
        parallelFor(
            exec,
            {0, lsView.b.size()},
            KOKKOS_LAMBDA(const size_t i) {
                lsView.b[i] = -lsView.b[i];
            }
        );
```